### PR TITLE
test(hx-error): unit test cases surf2006

### PR DESCRIPTION
**Pull request by lalithkarikelli**
## Description

Unit Test `<hx-error>`

image

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF 2006 (Component Test)

## Before you request a review for this PR:

- [x] Did you add unit tests for any new code if needed?
- [x] Did you run the unit tests via `yarn test` to ensure they all pass?
- [ ] Did you update the demo page and documentation if you changed/added functionality?
- [x] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] Have you linked to this PR from the Jira ticket(s)?

### CODE REVIEW

File: src/elements/hx-error/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { fixture, expect } from '@open-wc/testing';
```
Comment (100stacks) [inline src/elements/hx-error/index.spec.js:1]: Please see PR #685 for updated testing criteria.

File: src/elements/hx-error/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+/**
+ * <hx-error> component tests
+ *
+ * @type HXErrorElement
+ *
+ */
+describe('<hx-error> tests', () => {
```
Comment (100stacks) [inline src/elements/hx-error/index.spec.js:9]: Please add a screenshot of the test run.

Comment (lalithkarikelli) [reply src/elements/hx-error/index.spec.js:9]: Changes have been done, please review.

File: src/elements/hx-error/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            it('should have an unnamed <slot>', async () => {
+                const component = /** @type { HXErrorElement } */ await fixture(template);
+                const name = component.slot;
+
+                expect(name).to.be.equal('');
+            });
+        });
+
+        describe('verify Shadow DOM markup', () => {
+            it('markup should contain a #hxIconWrapper <span>', async () => {
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/elements/hx-error/index.spec.js]: Changes have been done, please review.

File: src/elements/hx-error/index.spec.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                const component = /** @type { HXErrorElement } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.getElementById(spanId);
+                const queryId = query.id;
+
+                expect(queryId).to.equal(spanId);
+            });
+        });
+    });
+});
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/elements/hx-error/index.spec.js]: Changes have been done, please review.

**Pull request merged**